flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -124,11 +126,33 @@ public void initializeState(FunctionInitializationContext context) throws Except
     if (context.isRestored()) {
       LOG.info("Restoring state for the {}.", getClass().getSimpleName());
       lastSnapshotId = lastSnapshotIdState.get().iterator().next();
+    } else if (scanContext.startTag() != null) {
+      Snapshot currentSnapshot =
+          scanContext.branch() == null
+              ? table.currentSnapshot()
+              : table.snapshot(scanContext.branch());
+      Preconditions.checkNotNull(
```

Comment:
I think it'll help if at some point (probably options/conf default) we set the default branch to main and always assume a branch is set in this code. Then we can just do table.snapshot(branch).

Nit on log message:

"No snapshots on branch %s in table %s"

Reply:
+1, can we actually do that in this PR? (can we check what is done in Spark, if we decided to go with using `main` branch or using if else? I cannot remember now)

Reply:
spark doesn't seem to use `Main` as the default.

My main concern is that `main` is not compatible with older version iceberg table.

Reply:
When we read the table metadata with the current Iceberg version we will set the main branch. https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableMetadataParser.java#L444 so it will be compatible with older version of tables.